infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java

```diff
@@ -106,7 +108,11 @@
     public LogUnitServer(ServerContext serverContext) {
         this.opts = serverContext.getServerConfig();
         this.serverContext = serverContext;
-        maxCacheSize = Utils.parseLong(opts.get("--max-cache"));
+        long maxCacheSizeFromUser = Utils.parseLong(opts.get("--max-cache"));
+
+        /* Ensure that maxCacheSize is at most CACHE_HEAP_RATIO * maxHeapSize */
```

Comment:
"at most" seems to be arbitrary. If a user specifies a cache size that should be used, otherwise use a default size which is the the cache heap ratio.